linux/test/app.test.js

```diff
@@ -3,41 +3,41 @@ const assert = require('assert');
 const path = require('path');
 
 const appPath = path.join(__dirname, '..', 'build', 'Headset-linux-x64', 'headset');
+let app = null;
 
 describe('application', function () {
   this.timeout(10000);
 
-  before(() => {
-    this.app = new Application({
+  before((done) => {
+    app = new Application({
       path: appPath,
       env: {
         DEBUG: 'headset*',
       },
     });
-    return this.app.start();
+    app.start()
+      .then(() => setTimeout(() => {
+        assert.equal(app.isRunning(), true, 'App is not running');
+        done();
+      }, 2000))
+      .catch(err => done(err));
   });
 
-  after(() => {
-    if (this.app && this.app.isRunning()) {
-      return this.app.stop();
-    }
-    throw new Error('App is not running');
-  });
+  after(() => app.stop());
 
-  it('start application', () => {
-    if (this.app && this.app.isRunning()) {
-      this.app.client.getWindowCount().then((count) => {
-        assert.equal(count, 2);
-      });
-    }
-    // Prints the logs produced only by DEBUG
-    this.app.client.getMainProcessLogs().then((logs) => {
+  // Tests that both windows were created
+  it('start application', () => app.client
+    .waitUntilWindowLoaded()
+    .getWindowCount()
+    .then(count => assert.equal(count, 2, 'Wrong number of windows'))
+    .then(() => app.client.getMainProcessLogs())
+    .then((logs) => {
       logs.forEach((log) => {
         log = log.split(' ');
         if (!Number.isNaN(Date.parse(log[0]))) {
           console.log('\t\x1b[33m%s \x1b[34m%s\x1b[0m', log[0], log[1], log.splice(2).join(' '));
         }
       });
-    });
-  });
+      return logs;
+    }));
 });
```

Comment:
would it be possible to have one test file? seems like both darwin and linux are very similar and they test the same things.

Reply:
We could have that, but if we decide to add other tests (i.e. media keys), they are going to be more specific to each OS.
We could have a single test file in the root that tests if Headset can start, and have any dedicated OS tests in each directory. Also, this might break how we run call the test to be run, if we introduce more test, we'll need multiple calls, instead of the single `npm run mocha` for all.

Reply:
yes we can have multiple test files. Some that are generic (e.g app can open) and some that are more specific to the os.

`mocha` can accept multiple arguments for directories to test.

so we can split it this way in the root `package.json`:
```json
"npm test:darwin": "mocha test/*.js darwin/test/*.js",
"npm test:linux": "mocha test/*.js linux/test/*.js",
"npm test:windows": "mocha test/*.js windows/test/*.js"
```
then travis can run `npm run test:$TRAVIS_OS_NAME`

That we can also remove mocha from the os dependencies and install it once .